crates/redeem/src/lib.rs

```diff
@@ -462,38 +521,94 @@ mod self_redeem {
             &vault_id.account_id,
         )?;
 
-        Pallet::<T>::release_replace_collateral(vault_id, consumed_issued_tokens)?;
         Ok(())
     }
 }
 // "Internal" functions, callable by code.
 #[cfg_attr(test, mockable)]
 impl<T: Config> Pallet<T> {
+    fn _request_replace(
+        old_vault_account: T::AccountId,
+        currency_pair: DefaultVaultCurrencyPair<T>,
+        amount: BalanceOf<T>,
+        new_vault_id: DefaultVaultId<T>,
+        griefing_currency: CurrencyId<T>,
+    ) -> Result<(), DispatchError> {
+        // Create VaultId for the old vault
+        let old_vault_id = VaultId::new(
+            old_vault_account.clone(),
+            currency_pair.collateral,
+            currency_pair.wrapped,
+        );
+
+        // Define old and new vault references
+        let old_vault = AccountOrVault::Vault(old_vault_id.clone());
+        let new_vault = AccountOrVault::Vault(new_vault_id.clone());
+
+        // probably this check is not strictly required, but it's better to give an
+        // explicit error rather than insufficient balance
+        ensure!(
+            old_vault_id.wrapped_currency() == new_vault_id.wrapped_currency(),
+            Error::<T>::InvalidWrappedCurrency
+        );
+
+        // don't allow vaults to replace themselves
+        ensure!(old_vault != new_vault, Error::<T>::ReplaceSelfNotAllowed);
+
+        // Calculate requestable tokens for old vault
+        let max_requestable_tokens: Amount<T> =
+            ext::vault_registry::requestable_to_be_replaced_tokens::<T>(&old_vault_id)?;
+        let requestable_tokens = amount.min(max_requestable_tokens.amount());
```

Comment:
I think this logic can be removed (from here & vault registry). `requestable_to_be_replaced_tokens` should now be equal to the redeemable amount, which should already be handled in `request_redeem`

Reply:
yaa it is handled in `redeem_request` -> `try_increase_to_be_redeemed_tokens` method, but it throws out an error if the amount is greater and does not take the minimum amount.